runtime/parachains/src/ump.rs

```diff
@@ -100,25 +103,30 @@ impl<XcmExecutor: xcm::latest::ExecuteXcm<C::Call>, C: Config> UmpSink for XcmSi
 			xcm::MAX_XCM_DECODE_DEPTH,
 			&mut &data[..],
 		)
-		.map(Xcm::<C::Call>::try_from);
+		.map(|xcm| {
+			(
+				Xcm::<C::Call>::try_from(xcm),
+				<C as Config>::WeightInfo::process_upward_message(data.len() as u32),
```

Comment:
This benchmark is a bit wrong IMO. `WeightInfo::process_upward_message` is measuring the execution of this entire code path, including the execution of the xcm itself. Yet, you are summing the weight used in the execution of the xcm with this weight. 

Ideally, the xcm operation itself should be something as close as possible to noop in the benchmark (`Transact` with `remark` might actually _be_ good for that) and then you could add them up.

That being said though, this all is kinda fine since you are over-couting the weight, which is less of a big deal.

Reply:
I do exactly this, the xcm operation is a Transact-remark with different sizes, to take the decoding effort into account but having a minimal xcm-execution overhead. https://github.com/paritytech/polkadot/blob/f88e7ee92e9c915139d70332c4e4f7e199b3499b/runtime/parachains/src/ump/benchmarking.rs#L55-L59

I can also just benchmark the xcm-decoding https://github.com/paritytech/polkadot/blob/f88e7ee92e9c915139d70332c4e4f7e199b3499b/runtime/parachains/src/ump.rs#L102

Or is there a even a benchmarked weight for this somewhere? If not it may make sense to add one and take decoding time into account in all places where it's used.

Reply:
Yeah this is really tricky. I am fine with it given

1. This is an overestimate and is therefore safe.
2. please make `// NOTE: ` and explain what is being done, and why it is not accurate.

Reply:
ideally, I would break down the function in ways that it can be easily weighed, but a story for another day.